.github/workflows/publish-image-on-release.yml

```diff
@@ -0,0 +1,50 @@
+name: Release Workflow
+
+on:
+  push:
+    tag:
+      - 'v*' #Match tags starting with 'v'
```

Comment:
I think this should be using the `release` action instead see https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release. Using the `push` event here will mean we end up having both workflows running at the same time when a tag is created. Also not sure whether the `push` event is triggered if the release is created using github.